### PR TITLE
feat(webapi): RDF API now supports HEAD calls similar to FDP API + added missing convience API calls

### PR DESCRIPTION
**Pull request by svandenhoek**
**IMPORTANT: This PR is build on https://github.com/molgenis/molgenis-emx2/pull/4126 as otherwise the tests failed no matter what. Until that PR is merged, this PR will remain a draft!**

What are the main changes you did:
- correct `content-type` is now returned for HEAD requests, similar to behaviour of `/api/fdp`:
https://github.com/molgenis/molgenis-emx2/blob/d5ac742d991409d2fa0f6d4f7ec621dd4fd085e5/backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/FAIRDataPointApi.java#L38-L39
https://github.com/molgenis/molgenis-emx2/blob/d5ac742d991409d2fa0f6d4f7ec621dd4fd085e5/backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/FAIRDataPointApi.java#L69-L71
- Fixed convenience APIs missing API points for row/column (f.e. `curl http://localhost:8080/pet%20store/api/jsonld/Pet` did nothing while `curl http://localhost:8080/pet%20store/api/rdf/Pet` worked)
- Reduced testing code duplicity.

how to test:
Should be covered already by unit-tests, but in case of manual testing:

| command line | tests | expected content-type |
| --- | --- | --- |
| `curl --head http://localhost:8080/pet%20store/api/rdf` | HEAD | text/turtle |
| `curl --head http://localhost:8080/pet%20store/api/ttl` | HEAD | text/turtle |
| `curl --head http://localhost:8080/pet%20store/api/jsonld` | HEAD | application/ld+json |
| `curl -H "Accept: application/ld+json" --head http://localhost:8080/pet%20store/api/rdf` | HEAD | application/ld+json |
| `curl -v http://localhost:8080/pet%20store/api/rdf \| grep "^>"` | GET | text/turtle |
| `curl -v http://localhost:8080/pet%20store/api/jsonld \| grep "^>"` | GET | application/ld+json |
| `curl -H "Accept: application/ld+json" -v http://localhost:8080/pet%20store/api/rdf \| grep "^>"` | GET | application/ld+json |

todo:
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

**Comment by svandenhoek** [comment]
Not sure why [backend/molgenis-emx2-webapi/build.gradle](https://github.com/molgenis/molgenis-emx2/pull/4134/files#diff-bfd67ffaec6df21494a3ca47d4d46bdd3c326c6b8ff665d859784a94268e7e30) is shown as changed because in edit mode/IDE it has a newline after `}`, but GitHub claims it got removed...

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java [hunk 1]
```diff
@@ -16,35 +16,107 @@
...
+    defineApiRoutes(
+        JSONLD_API_LOCATION,
+        RDFApi::jsonldHead,
+        RDFApi::jsonldForDatabase,
+        RDFApi::jsonldForSchema,
+        RDFApi::jsonldForTable,
+        RDFApi::jsonldForRow,
+        RDFApi::jsonldForColumn);
+  }
+
```
Comment (harmbrugge) [inline backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java:70]: Do you need to define al these handlers here?
Can't you send them to the same function in handle defineApiRoutes with an Type parameter and send them to the same function?:
`private static void defineApiRoutes(
      String apiLocation, RDFFormat rdfFormat)`
And then resolve that in the function handler?

**Review by harmbrugge — COMMENTED** [review]
Looks good but I think can be route definition can be simpeler

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java (hunk 1 above)
Comment (svandenhoek) [reply backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java:48]: Original code used extra functions for ttl/jsonld API so simply kept that to reduce code refactoring, though changed it now to your suggestion. Let me know what you think. Also added extra tests to validate on "if jsonld API would get a different content-type request will it still return jsonld"

**Review by harmbrugge — APPROVED** [review]
Code looks good, maybe you also want green light from @joerivandervelde functionally wise?

**Pull request merged**